src/ar_tx_replay_pool.erl

```diff
@@ -0,0 +1,290 @@
+	end;
+verify_tx(last_tx_in_mempool, TX, Diff, Height, FloatingWalletList, WeaveState, Mempool) ->
+	case ar_fork:height_1_8() of
+		H when Height >= H ->
+			%% Only verify after fork 1.8 otherwise it causes a soft fork
+			%% since current nodes can accept blocks with a chain of last_tx
+			%% references. The check would still fail on edge pre 1.8 since
+			%% TX is validated against a previous blocks' wallet list then.
+			case sets:is_element(TX#tx.last_tx, Mempool#mempool.tx_id_set) of
+				true ->
+					{invalid, last_tx_in_mempool};
+				false ->
+					verify_tx(
+						last_tx,
+						TX,
+						Diff,
+						Height,
+						FloatingWalletList,
+						WeaveState,
+						Mempool
+					)
+			end;
+		_ ->
+			verify_tx(
+				last_tx,
+				TX,
+				Diff,
+				Height,
+				FloatingWalletList,
+				WeaveState,
+				Mempool
+			)
+	end;
+verify_tx(last_tx, TX, Diff, Height, FloatingWalletList, WeaveState, Mempool) ->
+	case ar_tx:check_last_tx(FloatingWalletList, TX) of
+		true ->
+			NewMempool = Mempool#mempool {
+				tx_id_set = sets:add_element(TX#tx.id, Mempool#mempool.tx_id_set)
+			},
+			NewFWL = ar_node_utils:apply_tx(FloatingWalletList, TX, Height),
+			{valid, NewFWL, NewMempool};
+		false ->
+			verify_tx(anchor_check, TX, Diff, Height, FloatingWalletList, WeaveState, Mempool)
+	end;
+verify_tx(anchor_check, TX, Diff, Height, FloatingWalletList, WeaveState, Mempool) ->
+	case lists:member(TX#tx.last_tx, WeaveState#state.bhl) of
+		false ->
+			{invalid, tx_bad_anchor};
+		true ->
+			verify_tx(weave_check, TX, Diff, Height, FloatingWalletList, WeaveState, Mempool)
+	end;
+verify_tx(weave_check, TX, Diff, Height, FloatingWalletList, WeaveState, Mempool) ->
+	case weave_map_contains_tx(TX#tx.id, WeaveState#state.weave_map) of
+		true ->
+			{invalid, tx_already_in_weave};
+		false ->
+			verify_tx(mempool_check, TX, Diff, Height, FloatingWalletList, WeaveState, Mempool)
+	end;
+verify_tx(mempool_check, TX, _Diff, Height, FloatingWalletList, _WeaveState, Mempool) ->
+	case sets:is_element(TX#tx.id, Mempool#mempool.tx_id_set) of
+		true ->
+			{invalid, tx_already_in_mempool};
+		false ->
+			NewMempool = Mempool#mempool {
+				tx_id_set = sets:add_element(TX#tx.id, Mempool#mempool.tx_id_set)
+			},
+			NewFWL = ar_node_utils:apply_tx(FloatingWalletList, TX, Height),
+			{valid, NewFWL, NewMempool}
+	end.
+
+weave_map_contains_tx(TXID, WeaveMap) ->
+	lists:any(
+		fun(BH) ->
+			sets:is_element(TXID, maps:get(BH, WeaveMap))
+		end,
+		maps:keys(WeaveMap)
+	).
+
+apply_txs(TXs, Diff, Height, WalletList, WeaveState, Mempool) ->
+	lists:foldl(
+		fun(TX, {VerifiedTXs, FloatingWalletList, FloatingMempool}) ->
+			case verify_tx(
+				general_verification,
+				TX,
+				Diff,
+				Height,
+				FloatingWalletList,
+				WeaveState,
+				FloatingMempool
+			) of
+				{valid, NewFWL, NewMempool} ->
+					{VerifiedTXs ++ [TX], NewFWL, NewMempool};
+				{invalid, _} ->
+					{VerifiedTXs, FloatingWalletList, FloatingMempool}
```

Comment:
Am I correct in understanding that this fold just moves forward in case of an invalid transaction rather than halting